packages/node_modules/@node-red/nodes/core/function/89-delay.js

```diff
@@ -291,11 +291,12 @@ module.exports = function(RED) {
                     }
                 }
                 else if (!msg.hasOwnProperty("reset")) {
-                    if (maxKeptMsgsCount(node) > 0) {
+                    if (maxKeptMsgsCount(node) > 0 && !node.drop) {
```

Comment:
The `else` block we are in (starting line 293) is only entered if `node.drop` is set - see the `if` on line 230.
So adding `&& !node.drop` isn't doing anything here other than effectively adding `&& false` to the condition so the if block can never be entered.

That's a lot of words to say - if we think this is the fix, then it can be rewritten as just deleting the whole if block starting on line 294 down to 312.

Reply:
err yes. - removed